mod_wasm/modules/wasm/mod_wasm.c

```diff
@@ -204,67 +167,31 @@ static void *create_server_config(apr_pool_t *p, server_rec *s)
     cfg = (x_cfg *) apr_pcalloc(p, sizeof(x_cfg));
     cfg->local = 0;
     cfg->congenital = 0;
-    cfg->bEnableCGI = 0;
+    cfg->bWasmEnableCGI = 0;
     cfg->cmode = CONFIG_MODE_SERVER;
     /*
      * Note that we were called in the trace list.
      */
     sname = (sname != NULL) ? sname : "";
     cfg->loc = apr_pstrcat(p, "SVR(", sname, ")", NULL);
     trace_nocontext(NULL, __FILE__, __LINE__, sname);
-    return (void *) cfg;
-}
-
-/*
- * Post-config hook
- */
-static int post_config_hook(apr_pool_t *pconf, apr_pool_t *plog,
-                          apr_pool_t *ptemp, server_rec *s)
-{
-    trace_nocontext(NULL, __FILE__, __LINE__, "post_config_hook() - initializing wasm_runtime...");
-
-    // init wasm_runtime
-    const char* result = wasm_runtime_init_module();
-
-    if (strcmp(result, "") != 0)  // something went wrong
-        trace_nocontext(NULL, __FILE__, __LINE__, "post_config_hook() - ERROR! Couldn't initiale wasm_runtime!");
-    else
-        trace_nocontext(NULL, __FILE__, __LINE__, "post_config_hook() - wasm_runtime initialized!");
 
-    return_const_char_ownership(result);
+    // creates a new Wasm config for the current context
+    wasm_config_create(cfg->loc);
```

Comment:
If this returned an error, we should handle it here.

Reply:
I agree